addons/web/static/src/js/public/public_root.js

```diff
@@ -25,12 +25,14 @@ var localeDef = ajax.loadJS('/web/webclient/locale/' + lang.replace('-', '_'));
  * this Class instance. Its main role will be to retrieve RPC demands from its
  * children and handle them.
  */
-var PublicRoot = publicWidget.RootWidget.extend(ServiceProviderMixin, {
+var PublicRoot = publicWidget.RootWidget.extend({
+    env: publicEnv,
     events: _.extend({}, publicWidget.RootWidget.prototype.events || {}, {
         'submit .js_website_submit_form': '_onWebsiteFormSubmit',
         'click .js_disable_on_click': '_onDisableOnClick',
     }),
     custom_events: _.extend({}, publicWidget.RootWidget.prototype.custom_events || {}, {
+        'call_service': '_call_service',
```

Comment:
Component/widgets use this flow for service call, so they handle extension of global context with RPCs.
This doesn't seem the case with patched services in `env`... Not sure whether this is a problem or not?

(when entities make `rpc`, they rely on messaging service `rpc`, so in theory this affects almost all our RPCs in this refactoring)